### ISSUE TITLE
Summary gene table output is incorrect, only includes last entry for each gene from *genes.txt

### ISSUE THREAD

**Issue by mandel01** [mjmlab/pyinseq#85]
The `summary_gene_table.txt` output is not tabulating all of the relevant results from the *genes.txt files. In particular, the totals for the columns should be close to 1E6 (total of the counts per million, minus intergenic regions and regions excluded by `-d`). However, in some cases the total was much lower, ~15,000. It appears that instead of tabulating results in the summary gene table, only the last entry for each gene from each `*genes.txt` is being listed. For example:

SPL01_genes.txt:

```
contig	nucleotide	left_counts	right_counts	total_counts	cpm	three_primeness	locus_tag
CM001400	2279	11	6	17	3.611702084716934	0.996309963099631	VFSR5_0003
CM001400	4041	2	1	3	0.6373591914206354	0.4573643410852713	VFSR5_0004
CM001400	4919	6	2	8	1.6996245104550278	0.8577235772357723	VFSR5_0007
CM001400	4921	100	141	241	51.20118837745771	0.8550135501355014	VFSR5_0007
CM001400	5390	8	4	12	2.5494367656825414	0.21951219512195122	VFSR5_0007
CM001400	5833	28	5	33	7.010951105626989	0.5758928571428571	VFSR5_0008
CM001400	5872	6	5	11	2.336983701875663	0.5178571428571429	VFSR5_0008
CM001400	5907	11	5	16	3.3992490209100557	0.46577380952380953	VFSR5_0008
CM001400	6071	2	3	5	1.0622653190343923	0.22172619047619047	VFSR5_0008
CM001400	6176	6	1	7	1.4871714466481492	0.06547619047619048	VFSR5_0008
```

summary_gene_table:

Synonym | Code | COG | Product | SPL01
-- | -- | -- | -- | --
VFSR5_0001 | - | - | FMN-binding protein MioC | 0
VFSR5_0002 | - | - | tRNA modification GTPase TrmE | 0
VFSR5_0003 | - | - | membrane protein insertase | 3.61170208
VFSR5_0004 | - | - | YidD | 0.63735919
VFSR5_0005 | - | - | ribonuclease P | 0
VFSR5_0006 | - | - | 50S ribosomal protein L34 | 0
VFSR5_0007 | - | - | cystine transport ATP-binding protein | 2.54943677
VFSR5_0008 | - | - | cystine transport system permease | 1.48717145

Note that the genes with no entries or only one entry are correct. However, VFSR5_0007 should sum the total from the 3 entries (1.70, 51.20, 2.55); instead only 2.55 is reported in the table. Similar for VFSR5_0008.

**Comment by mandel01** [comment]
Here is a [workaround using pandas](https://gist.github.com/mandel01/65869364b901eaca41421184413e7d6c) for the `-d = 0` case.

**Comment by eburgoswisc** [comment]
Also fix test files, `summary_gene_table.txt` has the same bug. It is only grabbing the last entry from multiple gene mappings in `_genes.txt`.

**Comment by mandel01** [comment]
Looking to get this bug resolved...

In the old version, `process_mapping.py / map_genes()` returned a `sample_dict` that was the aggregated insertions for each gene, filtered on the 5'-3' threshold that the user had set with the `-d` parameter. This information was fed into the `process_mapping.py / build_gene_table()` function. 

In the new snakemake version,  I don't think this information reaches `process_mapping.py / build_gene_table()`. I'm still figuring out snakemake...but it looks like a similarly-named `samples_dict` from settings is instead passed for this parameter.

Consistent with this being the problem, the current version is also not properly considering three_primeness. Below from `test_pyinseq_script`.

ES001_02_genes.txt
```
contig	nucleotide	left_counts	right_counts	total_counts	cpm	three_primeness	locus_tag
CP000020	11152	1	7	8	29197.080291970804	0.04094292803970223	VF_0012
CP000020	13423	7	7	14	51094.890510948906	0.9801488833746899	VF_0012
CP000020	39761	14	14	28	102189.78102189781	0.21212121212121213	VF_0033
CP000021	566179	21	21	42	153284.67153284673	0.9682971014492754	VF_A0492
CP000021	567228	14	14	28	102189.78102189781	0.018115942028985508	VF_A0492
CP000022	3788	7	7	14	51094.890510948906	0.963855421686747	VF_B0007
CP000022	3971	7	7	14	51094.890510948906	0.5963855421686747	VF_B0007
CP000022	4247	28	28	56	204379.56204379562	0.04216867469879518	VF_B0007
```


summary_gene_table.txt
```
CP000020	11054	13471	+	2418	AAW84507.1	gyrB	VF_0012	-	-	DNA gyrase, subunit B	0	51094.890510948906
```

Thoughts @eburgoswisc ?

**Comment by mandel01** [comment]
Here is the [summary_gene_table.txt](https://github.com/mjmlab/pyinseq/blob/fe3a0fa5270e707a2cfade7dd637ae25f0cdd03a/pyinseq/tests/data/output_pyinseq/summary_gene_table.txt) from fe3a0fa5270e707a2cfade7dd637ae25f0cdd03a, which should be useful for testing.

**Comment by mandel01** [comment]
Fixed by b85809b42cf81058688a5f39ab4ab4e260a14e41.

Thanks @eburgoswisc!
